src/providers/wfs/qgsoapifcollection.cpp

```diff
@@ -0,0 +1,497 @@
+/***************************************************************************
+    qgsoapifcollection.cpp
+    ---------------------
+    begin                : October 2019
+    copyright            : (C) 2019 by Even Rouault
+    email                : even.rouault at spatialys.com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <nlohmann/json.hpp>
+using namespace nlohmann;
+
+#include "qgslogger.h"
+#include "qgsoapifcollection.h"
+#include "qgsoapifutils.h"
+
+#include <set>
+
+#include <QTextCodec>
+
+bool QgsOapifCollection::deserialize( const json &j )
+{
+  if ( !j.is_object() )
+    return false;
+  const char *idPropertyName = "id";
+  if ( !j.contains( "id" ) )
+  {
+#ifndef REMOVE_SUPPORT_DRAFT_VERSIONS
+    if ( j.contains( "name" ) )
+    {
+      idPropertyName = "name";
+    }
+    else
+#endif
+    {
+      QgsDebugMsg( QStringLiteral( "missing id in collection" ) );
+      return false;
+    }
+  }
+  const auto id = j[idPropertyName];
+  if ( !id.is_string() )
+    return false;
+  mId = QString::fromStdString( id.get<std::string>() );
+
+  mLayerMetadata.setType( QStringLiteral( "dataset" ) );
+
+  const auto links = QgsOAPIFJson::parseLinks( j );
+  const auto selfUrl = QgsOAPIFJson::findLink( links,
+                       QStringLiteral( "self" ),
+                       QStringList() <<  QStringLiteral( "application/json" ) );
```

Comment:
nothing really important, but you could also do
```suggestion
                       { QStringLiteral( "application/json" ) } );
```

Reply:
Taken into account